packages/rum-core/src/browser/cookieObservable.spec.ts

```diff
@@ -0,0 +1,58 @@
+import type { Subscription } from '@datadog/browser-core'
+import { ONE_MINUTE, STORAGE_POLL_DELAY, deleteCookie, setCookie } from '@datadog/browser-core'
+import type { Clock } from '@datadog/browser-core/test'
+import { mockClock } from '@datadog/browser-core/test'
+import type { RumConfiguration } from '../domain/configuration'
+import { createCookieObservable } from './cookieObservable'
+import type { CookieChange, CookieObservable } from './cookieObservable'
+
+const COOKIE_NAME = 'cookie_name'
+const COOKIE_DURATION = ONE_MINUTE
+
+describe('cookieObservable', () => {
+  let observable: CookieObservable
+  let subscription: Subscription
+  let originalSupportedEntryTypes: PropertyDescriptor | undefined
+  let clock: Clock
+  beforeEach(() => {
+    clock = mockClock()
+    observable = createCookieObservable({} as RumConfiguration, COOKIE_NAME)
+    originalSupportedEntryTypes = Object.getOwnPropertyDescriptor(window, 'cookieStore')
+  })
+
+  afterEach(() => {
+    subscription?.unsubscribe()
+    if (originalSupportedEntryTypes) {
+      Object.defineProperty(window, 'cookieStore', originalSupportedEntryTypes)
+    }
+    clock.cleanup()
+    deleteCookie(COOKIE_NAME)
+  })
+
+  it('should notify observers on cookie change', (done) => {
+    subscription = observable.subscribe((cookieChange) => {
+      expect(cookieChange).toEqual({
+        name: COOKIE_NAME,
+        value: 'foo',
+      })
+
+      done()
+    })
+    setCookie(COOKIE_NAME, 'foo', COOKIE_DURATION)
+    clock.tick(STORAGE_POLL_DELAY)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏any reason to be related to session store?
```suggestion
    clock.tick(WATCH_COOKIE_INTERVAL_DELAY)
```